app/script/view_model/content/InputBarViewModel.js

```diff
@@ -114,6 +114,8 @@ z.viewModel.content.InputBarViewModel = class InputBarViewModel {
     });
 
     this.richTextInput = ko.pureComputed(() => {
+      this.updateSelectionState();
```

Comment:
This side effect doesn't belong here in a pureComputed. What is it achieving?

Reply:
It updates the internal mention ranges (in `this.currentMentions`) . As I was told by @AndyLnd this is needed to ensure that the mentions are up-to-date and the correct beginnings and endings of the mentions are saved.